### ISSUE TITLE
SHAP in multiclass classification.

### ISSUE THREAD

**Issue by JuanCorp** [shap/shap#71]
I'm using SHAP for a multiclass classification problem, with 4 classes. The input uses 135 features, but the SHAP contributions matrix outputs 544 features. I assume this is  135 features * 4 classes + 4 biases, right? So if I wanted to know the  contributions for a particular class, I would access them like this: 

`class = 0 `
`class_contribs = contribs[class * n_features:(class + 1) * n_features]`

Is this correct?

**Comment by slundberg** [comment]
Hmm...could you check what version of shap you are using? I thought a while back I made all multitask settings output a list of contribution matrices, one for each task (in an effort to avoid this confusion). If you have the latest perhaps a brief code snippet could help.

**Comment by JuanCorp** [comment]
I am using version 0.14.1. Here's a snippet: 

`clf = xgb.XGBClassifier(**train_params)`
`clf.fit(X,y) #X.shape is (134132, 135)` 
`model = clf.get_booster()`
`contribs = model.predict(xgb.DMatrix(X),pred_contribs=True,approx_contribs=True)`
`contribs.shape #contribs.shape is (134132, 544)`

Here are the training params 
`train_params = {'grow_policy': 'lossguide',
 'tree_method': 'hist',
 'colsample_bytree': 0.5133672635483673,
 'gamma': 3.7708645374614003,
 'learning_rate': 0.09376663157182812,
 'max_depth': 11,
 'min_child_weight': 71,
 'n_estimators': 1572,
 'objective': 'multi:softmax',
 'random_state': 777,
 'reg_alpha': 5.564900228441193,
 'reg_lambda': 9.467983053780042,
 'subsample': 0.8105012933433529}`

Maybe it's because I'm using the sklearn interface for XGBClassifier?

**Comment by slundberg** [comment]
Ah! No it's because you are using XGBoost directly and not through the TreeExplainer interface. I also realized I didn't update TreeExplainer to handle the sklearn interface or multitask output from XGBoost ( I just pushed support for that now). The other issue is that the newest version of XGBoost produces a matrix with 3 dims for multitask output, but the older version you have only makes two. I can't recall how the older one ordered the rows, but I think reshape(N, Class, feature) should do it. You can check and make sure you got it right by ensuring the last feature column is constant (since it is the bias). Let me know if that works.

**Comment by JuanCorp** [comment]
Oh right, I'm using an older version of XGBoost, it's not a pip one, I think it's the January version of the package at Github. Haven't updated it since. Your reshape fix worked, but I'll also update XGBoost. Thanks!

**Comment by slundberg** [comment]
Great!
